Content.Shared/Climbing/SharedBonkSystem.cs

```diff
@@ -0,0 +1,59 @@
+using Content.Shared.Interaction;
+using Content.Shared.Stunnable;
+using Content.Shared.CCVar;
+using Content.Shared.Damage;
+using Content.Shared.DragDrop;
+using Robust.Shared.Configuration;
+using Content.Shared.Popups;
+
+namespace Content.Shared.Climbing;
+
+public sealed class SharedBonkSystem : EntitySystem
+{
+    private const string BonkMessage = "Bonk!";
+
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+    [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
+    [Dependency] private readonly SharedStunSystem _stunSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<BonkableComponent, DragDropEvent>(OnDragDrop);
+    }
+
+    public bool TryBonk(EntityUid user, EntityUid climbable)
+    {
+        BonkableComponent? bonkableComponent = null;
+        if (Resolve(climbable, ref bonkableComponent))
+        {
+            if (!_cfg.GetCVar(CCVars.GameTableBonk))
+            {
+                // Not set to always bonk, try clumsy roll.
+                if (!_interactionSystem.TryRollClumsy(user, bonkableComponent.BonkClumsyChance))
+                    return false;
+            }
+
+            // BONK!
+
+            _popupSystem.PopupEntity(BonkMessage, user, PopupType.Medium);
```

Comment:
get the message with `Loc.GetString()` once you do the previous review and make two popups, one for yourself and one for others, here is one example from `DrinkSystem` (the identity part is important in this case)
```cs
                var targetName = Identity.Entity(uid, EntityManager);
                var userName = Identity.Entity(args.User, EntityManager);

                _popupSystem.PopupEntity(
                    Loc.GetString("drink-component-force-feed-success", ("user", userName), ("flavors", flavors)), uid, uid);

                _popupSystem.PopupEntity(
                    Loc.GetString("drink-component-force-feed-success-user", ("target", targetName)),
                    args.User, args.User);
```